content/en/platform/corda/5.0-dev-preview-2/getting-started/fast-feedback-with-the-simulator/kotlin-flow-test-code.md

```diff
@@ -0,0 +1,83 @@
+1. Set up Alice and Bob identities:
+   ```kotlin
+   MyFirstFlowTest {
+       // Names picked to match the corda network in config/dev-net.json
+       private val aliceX500 = MemberX500Name.parse("CN=Alice, OU=Test Dept, O=R3, L=London, C=GB")
+       private val bobX500 = MemberX500Name.parse("CN=Bob, OU=Test Dept, O=R3, L=London, C=GB")
+   ```
+2. Declare the test method in the standard way for JUnit test, with a `@Test` annotation:
+   ```kotlin
+       @Test
+       fun `test that MyFirstFLow returns correct message`() {
+   ```      
+3. Instantiate a `Simulator` class:
+   ```kotlin
+   // Instantiate an instance of the Simulator
+   val simulator = Simulator()
+   ```
+   To find out more about this class, read the [README.md from the source code](https://github.com/corda/corda-runtime-os/blob/release/os/5.0/simulator/README.md).
+
+4. Convert the `MemberX500Name` for Alice and Bob into `HoldingIdentities` using the `HoldingIdentity` static method. The `HoldingIdentity` in Corda holds the `MemberX500Name` and the `GroupId` which is the unique identifier for the application network. However, the class used here is the simulated version provided by Simulator; that is `net.corda.simulator.HoldingIdentity` rather than `net.corda.virtualnode.HoldingEntity`.
+   ```kotlin
+   // Create Alice's and Bob HoldingIDs
+   val aliceHoldingID = HoldingIdentity.Companion.create(aliceX500)
+   val bobHoldingID = HoldingIdentity.Companion.create(bobX500)
+   ```        
+5. Use Simulator to create virtual nodes for Alice and Bob from their respective holding identities. The second argument allows you to specify which flows should be loaded up to each virtual node. In this case, Alice runs the initiating flow, with Bob running the responder flow.
+   ```kotlin
+   // Create Alice and Bob's virtual nodes, including the classes of the flows which will be registered on each node.
+   val aliceVN = simulator.createVirtualNode(aliceHoldingID, MyFirstFlow::class.java)
+   simulator.createVirtualNode(bobHoldingID, MyFirstFlowResponder::class.java)
+   ```
+    You can read more about initiating flows and responder flows in the section on [Your first flow ](../first-flow.html#initiating-and-responding-flows).
+6. Create the arguments to pass to the flow. In the flow file `MyFirstFlow.kt`, create a class `MyFirstFlowStartArgs` specifically for holding the flow start arguments:
+   ```kotlin
+   // A class to hold the arguments required to start the flow
+   class MyFirstFlowStartArgs(val otherMember: MemberX500Name, val message: String)
+   ```
+   Use the same class to specify that the message needs to go to Bob and the message should be "Hello Bob":
```

Comment:
This is wrong I think. 
We do not use the same class for message from Alice to Bob. We do not need to specify that for the test.